blog: add blog for imagevalidating-policy (#2040)
kyverno/website: src/content/blog/imagevalidating-policy/index.md

---
date: 2026-05-26
title: ImageValidatingPolicy - Trust What Your Cluster Actually Runs
tags:
  - General
authors:
  - name: Kirti Goyal
excerpt: A practical guide to ImageValidatingPolicy in Kyverno like verifying container image signatures, checking attestations, and enforcing supply chain security before anything runs in the cluster.
draft: false
---

> The other policy covered so far deals with Kubernetes resource configuration.
> Labels, resource limits, namespaces, cleanup schedules. All of it is about how
> resources are configured.

Reviewer: **suggestion**

@kirti763  if this intended to mention vpol, mpol, gpol etc. I think using plural policies might be better?

```suggestion
The other policies covered so far deal with Kubernetes resource configuration. Labels, resource limits, namespaces, and cleanup schedules are all about how resources are configured.
```

**ImageValidatingPolicy** deals with something completely different.

> Not configuration. **Trust**.

Reviewer: ```suggestion
Not configuration. but **Trust**.
```

> When a Pod is scheduled and Kubernetes pulls a container image like how does the cluster
> know that image is actually what it claims to be? How does it know the image wasn't
> tampered with after it was built? How does it know it came from the CI pipeline and
> not from somewhere else entirely?

Reviewer: **suggestion**

```suggestion
When a Pod is scheduled and Kubernetes pulls a container image, how does the cluster
verify that image is actually what it claims to be? How does it ensure the image wasn't
tampered with after it was built? How does it know it came from the CI pipeline and
not from somewhere else entirely?
```
